types/ensure/package.pp

```diff
@@ -0,0 +1,2 @@
+# @summary Validate the value of the ensure parameter for a package
+type Stdlib::Ensure::Package = Variant[Enum['present', 'absent', 'purged', 'disabled', 'installed', 'latest'], Pattern[/\./]]
```

Comment:
When I was working on this I was also thinking "this would not allow non decimal numbers or bare git commit hash as version", but the Puppet docs don't allow it, so I'm also not allowing it here :)

Reply:
Or, am I misreading https://puppet.com/docs/puppet/7/types/package.html#package-attribute-ensure, and is that actually a regex wildcard . in there, `/./`, in that case, my version is too strict (and `String[1]` would be better).

Reply:
That's indeed a regex equivalent to `String[1]`.

Reply:
I changed it to `String[1]`